include/crab/domains/fixed_tvpi_domain.hpp

```diff
@@ -213,6 +213,26 @@ class fixed_tvpi_domain
     }
   }
 
+  boost::optional<number_t> is_var_constant(const variable_t &x) {
+    interval_t ival = m_base_absval.at(x);
+    return ival.singleton();
+  }
+
+  void rewrite_apply_var(arith_operation_t op, const variable_t &x,
+                         const variable_t &y, const variable_t &z,
+                         unsigned coefficient) {
+    assert(coefficient > 1);
+
+    if (boost::optional<number_t> n = is_var_constant(z)) {
+      rewrite_apply(op, x, y, *n, coefficient);
+    } else if (boost::optional<number_t> n = is_var_constant(y)) {
+      rewrite_apply(op, x, z, *n, coefficient);
```

Comment:
This assumes that `op` is commutative so it won't work for subtraction and division.

Reply:
Yes, you are right. If `y` is some constant `n`, then we should have a special rewrite for `x := n - z` or `x := n / z`. How about adding constraints for those two cases?

Reply:
Unless we need those cases now I would just exclude them (i.e., do nothing if sub or div)

Reply:
Ok. Let me add one condition based on this.

Reply:
DONE.